### PR TITLE
Out of memory error on making multiple requests

### PR DESCRIPTION
**Pull request by aby040**
Fixes: #374 

Creating a separate axios instance and moving the interceptor code out of the sendRequest() method.

An interceptor was added to the static axios instance on each sendRequest() call which can lead to out of memory issue if hundreds of requests are made as the added interceptor is not ejected after the request is made.

### CODE REVIEW

File: lib/axiosHttpClient.ts [hunk 1]
```diff
@@ -16,6 +16,18 @@ import * as http from "http";
...
+
+// This hack is still required with 0.19.0 version of axios since axios tries to merge the
+// Content-Type header from it's config["<method name>"] where the method name is lower-case,
+// into the request header. It could be possible that the Content-Type header is not present
+// in the original request and this would create problems while creating the signature for
+// storage data plane sdks.
+client.interceptors.request.use((config: AxiosRequestConfig) => ({
+  ...config,
+  method: (config.method as Method) && (config.method as Method).toUpperCase() as Method
+}));
```
Comment (amarzavery) [inline lib/axiosHttpClient.ts]: This change LGTM.

File: lib/policies/msRestUserAgentPolicy.browser.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@
 import { TelemetryInfo } from "./userAgentPolicy";
 
 interface NavigatorEx extends Navigator {
-  readonly oscpu: string | undefined;
+  readonly oscpu: string;
```
Comment (amarzavery) [inline lib/policies/msRestUserAgentPolicy.browser.ts]: @aby040 - How is this change related to the current problem being fixed?

File: .scripts/unit.ts [hunk 3]
```diff
@@ -75,13 +75,13 @@ function stopTestServer(testServer: ServerProcess): void {
...
   console.log(`Running Node.js Unit Tests...`);
-  return executeSync(`nyc mocha`, repositoryRootFolderPath).status;
+  return executeSync(`nyc mocha`, repositoryRootFolderPath).status || 0;
 }
 
 function runBrowserUnitTests(): number {
   console.log(`Running Browser Unit Tests...`);
   const portNumber: string | number = process.env.PORT || 3001;
-  return executeSync(`${mochaChromeFilePath} http://localhost:${portNumber} --timeout 60000`, repositoryRootFolderPath).status;
+  return executeSync(`${mochaChromeFilePath} http://localhost:${portNumber} --timeout 60000`, repositoryRootFolderPath).status || 0;
```
Comment (amarzavery) [inline .scripts/unit.ts]: @aby040 -  Why are you defaulting the exit code to `0`? How is this related to the current problem being fixed in the PR?

File: lib/policies/msRestUserAgentPolicy.browser.ts (hunk 2 above)
Comment (aby040) [reply lib/policies/msRestUserAgentPolicy.browser.ts]: Sorry, this was accidentally added to this merge request. This change was made to fix some build errors.

File: .scripts/unit.ts (hunk 3 above)
Comment (aby040) [reply .scripts/unit.ts]: Sorry, this was accidentally added to this merge request. This change was made to fix some build errors.

File: lib/axiosHttpClient.ts (hunk 1 above)
Comment (aby040) [reply lib/axiosHttpClient.ts]: @amarzavery  Can we have this added to latest 1.x.x release so that the memory leak issue can be resolved in `@azure` packages which are still relying on versions `^1.8.1` of `@azure/ms-rest-js`

**Comment by daviwil** [comment]
Hi @aby040, thanks for this change!  Can you turn on "Allow edits from maintainers" on this PR so that I can make a couple of tweaks before merging?

File: lib/axiosHttpClient.ts (hunk 1 above)
Comment (amarzavery) [reply lib/axiosHttpClient.ts]: Yes. If everything goes well, then the next version in the 1.x release train would be 1.8.14. @daviwil is the current owner and should be able to provide more details on this.

**Comment by princjef** [comment]
fwiw it looks like the build failures are due to a newer version of typescript being pulled in by the build than the version used when the original code was written. Probably a side effect of the package-lock.json file being excluded from the repo.

**Comment by bterlson** [comment]
Overall change looks good once we can fix the build issue (probably not an issue in azure-sdk-for-js as we're using a lockfile there locked to node v8 types - a similar thing could be considered for this repo).

File: lib/axiosHttpClient.ts (hunk 1 above)
Comment (aby040) [reply lib/axiosHttpClient.ts]: Thanks @amarzavery

**Comment by aby040** [comment]
Sure @daviwil, I have turned on "Allow edits from maintainers". Please let me know if anything else has to be added from my end for this to get merged.

**Comment by aby040** [comment]
@bterlson, sorry I didnt get what you meant by 

> using a lockfile there locked to node v8 types

Shall I make the changes in `.scripts/units.ts` and `lib/policies/msRestUserAgentPolicy.browser.ts` to fix the type issues?

**Comment by princjef** [comment]
Some clarification on next steps would be good. Is the desire to fix the existing build/lockfile issues via a separate issue/PR or to do it as a part of this change?

@bterlson thoughts?

**Comment by bterlson** [comment]
Sorry for the confusing unrelated comment. From my perspective, @aby040's proposal to update .scripts/unit.ts and lib/policies/msRestUserAgentPolicy.browser.ts based on the latest node types seems appropriate, but I'd defer to either @daviwil or @amarzavery

**Comment by princjef** [comment]
@daviwil I've created #377 to try to address the build issues as a standalone change to unblock this PR. I'm also fine if you instead want to do the code changes that were previously a part of this PR. Any thoughts on how you want to proceed so that we can resolve this issue?

**Comment by daviwil** [comment]
I integrated @princjef's build and CI fixes from #377 and fixed the remaining test issues caused by the interceptor move, so this is ready to be merged.  Thanks to @aby040 and @princjef for the help!

**Comment by daviwil** [comment]
`ms-rest-js` 1.8.14 is out: https://www.npmjs.com/package/@azure/ms-rest-js/v/1.8.14
